### DISCUSSION TITLE
Questions about MXPredCreateMultiThread

### DISCUSSION THREAD

**Discussion by Thelta** [apache/incubator-mxnet#13457] [Q&A]
Hello, I have been trying trying to use MXPredCreateMultiThread on my project. There are few questions I have.

1. I have [asked](https://discuss.mxnet.io/t/reusing-mxpredcreatemultithreads-predicthandles/1944) about using reusing PredictHandles. For those who didn't say clicked the link, short answer is it fine as long as it is used in same thread. Will it be possible to reuse it in different threads at later time or is there a technical problem that prevents it?

2. Is it plausible to reshape the handles created by MXPredCreateMultiThread? I have been using MXPredReshape and I have been observing memory leaks in my project. After I changed MXPredCreate memory leaks ended. I wonder is it because of reshaping?

**Comment by vrakesh** [comment]
@Thelta Thank you for your questions
1. Regarding this,
     There was an answer in the forum 3 days ago, perhaps that answers your question?

     https://discuss.mxnet.io/t/reusing-mxpredcreatemultithreads-predicthandles/1944

     Note: That MXNet is not thread safe, PredictHandles created should be fine as long as they are in the same thread.

2. This one I am not sure, I will let someone else answer this.

**Comment by vrakesh** [comment]
@mxnet-label-bot add [Question]
